CHANGELOG.md

```diff
@@ -6,454 +6,473 @@ All notable changes to this project will be documented in this file.
 
 ## Added
 
-- Automate
-  - Import Rake task `OVERWRITE` argument: Setting  `OVERWRITE=true` removes the target domain prior to import
-  - New `extend_retires_on` method: Used by Automate methods to set a retirement date to specified number of days from today, or from a future date.
-  - Service model updates
-    - MiqAeServiceHardware
-    - MiqAeServicePartition
-    - MiqAeServiceVolume
+### Automate
+
+- New methods
+  - `extend_retires_on` method: Used by Automate methods to set a retirement date to specified number of days from today, or from a future date.
+  - `taggable?` to programmatically determine if a Service Model class or instance is taggable.
+- Service model updates
+  - MiqAeServiceHardware
+  - MiqAeServicePartition
+  - MiqAeServiceVolume
+- Generic Objects: Model updates
+  - Associations
+  - Tagging
+  - Service Methods: `add_to_service / remove_from_service`
+  - Service models created for GenericObject and GenericObjectDefinition
+  - Process Generic Object method call via automate
+  - Methods content stored in Automate
+  - Generic Object Definition model contains the method name only (Parameters defined in automate)
+  - Methods can return data to caller
+  - Methods can be overridden by domain ordering
+- Git Automate support
+  - Branch/Tag support
+  - Contents are locked and can be copied to other domains for editing
+  - Editable properties
+    - Enabled/Disabled
+    - Priority
+    - Removal of Domain
+    - Dedicated Server Role to store the repository
+-  Import Rake task `OVERWRITE` argument: Setting  `OVERWRITE=true` removes the target domain prior to import
+- Added /System/Process/MiqEvent instance
+- Added Provider refresh call to Amazon retire state machine in Pre-Retirement state
+- Class Schema allows for EMS, Host, Policy, Provision, Request, Server, Storage, and VM(or Template) datatypes
+  - Value is id of the objects
+  - If object is not found, the attribute is not defined.
+- Null Coalescing Operator
+  - Multiple String values separated by “||”
+  - Evaluated on new attribute data type “Null Coalescing”
+  - Order dependent, left to right evaluation
+  - First non-blank value is used
+  - Skip and warn about missing objects
+- Service Dialogs
+  - Added ‘Visible’ flag to all dialog fields
+  - Support for “visible” flag for dynamic fields
+- Expose Compliance and ComplianceDetail models
+- New associations on VmOrTemplate and Host models:
+  - `expose :compliances`
+  - `expose :last_compliance`
+- New Service Models
+  - Compliance: `expose :compliance_details`
+  - ComplianceDetail: `expose :compliance`, `expose :miq_policy`
+  - Service Model: Expose provision_priority value
+- Engine: Allow arguments in method calls during substitution
+   operation
+- Retirement
+  - Restored retirement logic to verify that VM was provisioned or contains Lifecycle tag before processing
+  - Built-in policy to prevent retired VM from starting on a resume power
+- Added lifecycle tag as a default tag
```

Comment:
This item can be indented under the `Retirement` heading since it was added for that purpose.